mesonbuild/dependencies/base.py

```diff
@@ -355,6 +363,28 @@ def generate_link_whole_dependency(self) -> Dependency:
         new_dep.libraries = []
         return new_dep
 
+    def get_as_static(self, recursive: bool) -> Dependency:
+        new_dep = copy.deepcopy(self)
```

Comment:
I think a shallow copy is enough here, you only modify the object itself, not what it contains. When recusing that would do a shallow copy of contained InternalDependency as well. I think you copied that deepcopy() from other methods that probably shouldn't do that either.